File: MatlabOctaveCodeSnippets/snip_transforms_freqz.m
N1=5; %num. non-zero samples in the (aperiodic) pulse
x=[ones(1,N1) zeros(1,2)]; %signal x[n]
halfN=512; N=2*halfN; %half of the DFT dimension and N
k=0:N-1; %indices of freq. components
wk = k*2*pi/N; %angular frequencies
Xk_fft=fft(x,N);%use 2N-DFT to get N positive freqs.
Xk_fft=Xk_fft(1:halfN);%discard negative freqs.
wk=(0:halfN-1)*(2*pi/N); %positive freq. grid
subplot(211); plot(wk,20*log10(abs(Xk_fft))); %in dB
subplot(212); plot(wk,angle(Xk_fft)*180/pi); %in degrees

File: PythonCodeSnippets/snip_transforms_freqz.py
import numpy as np
from numpy.fft import fft
import matplotlib.pyplot as plt

N1 = 5  # num. non-zero samples in the (aperiodic) pulse
x = np.concatenate((np.ones(N1), np.zeros(2)))  # Signal x[n]
halfN = 512
N = 2 * halfN  # Half of the DFT dimension and N
k = np.arange(N)  # Indices of freq. components
wk = k * 2 * np.pi / N  # Angular frequencies
Xk_fft = fft(x, N)  # Use 2N-DFT to get N positive freqs.
Xk_fft = Xk_fft[0:halfN]  # Discard negative freqs.
wk = (np.arange(halfN)) * (2 * np.pi / N)  # Positive freq. grid

plt.subplot(211)
plt.plot(wk, 20 * np.log10(abs(Xk_fft)))  # In dB
plt.subplot(212)
plt.plot(wk, np.angle(Xk_fft) * 180 / np.pi)  # In degrees
plt.show()
